rpcserver/router.go

```diff
@@ -449,90 +492,99 @@ func (server *routedBoltzServer) createSwap(ctx context.Context, isAuto bool, re
 		return nil, handleError(err)
 	}
 
-	response, err := server.boltz.CreateSwap(createSwap)
+	if swapResponse == nil {
```

Comment:
Not a huge fan of this. Could we split the logic of this function into multiple ones for the different use cases it has now?

Reply:
its a bit hard since it relies on the request and some other variables defined above (like `wallet` etc.)
pulling the routing hint stuff into its own function wouldnt be an issue tough

Reply:
Then let's start with that. I don't like that this function in the router keeps growing